includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1041,20 +1042,28 @@ public function process_payment_for_order( $cart, $payment_information, $additio
 
 			$payment_methods = WC_Payments::get_gateway()->get_payment_method_ids_enabled_at_checkout( null, true );
 
-			// Create intention, try to confirm it & capture the charge (if 3DS is not required).
-			$intent = $this->payments_api_client->create_and_confirm_intention(
-				$converted_amount,
-				$currency,
-				$payment_information->get_payment_method(),
-				$customer_id,
-				$payment_information->is_using_manual_capture(),
-				$save_payment_method,
-				$metadata,
-				$this->get_level3_data_from_order( $order ),
-				$payment_information->is_merchant_initiated(),
-				$additional_api_parameters,
-				$payment_methods
-			);
+			// phpcs:ignore
+			if ( ! empty( $_POST['platform-checkout-intent'] ) ) {
+				// If the intent is included in the request use that intent.
+				// phpcs:ignore
+				$intent_id = wp_unslash( $_POST['platform-checkout-intent'] );
+				$intent    = $this->payments_api_client->get_intent( $intent_id );
```

Comment:
Yeah, maybe `sanitize_key` 👍.

Reply:
Ah, turns out sanitize_key won't work since it filters out upper case characters and Stripe IDs may have upper case characters 😩

I think [sanitize_user( $intent_id, true /* enable strict mode */ )](https://developer.wordpress.org/reference/functions/sanitize_user/) is more appropriate since it will not replace upper case characters 🤔 

Made that change in df263b3e96c4c6f078a16a5ac52c1aaf31b71937